src/components/app/details/cIDetails/CIDetails.tsx

```diff
@@ -668,58 +672,22 @@ export const NoVulnerabilityViewWithTool = ({ scanToolId }: { scanToolId: number
     )
 }
 
-const SecurityTab = ({ ciPipelineId, artifactId, status, appIdFromParent }: SecurityTabType) => {
-    const [isCollapsed, setIsCollapsed] = useState(false)
-    const [securityData, setSecurityData] = useState({
-        vulnerabilities: [],
-        lastExecution: '',
-        severityCount: {
-            critical: 0,
-            high: 0,
-            medium: 0,
-            low: 0,
-            unknown: 0,
-        },
-        scanEnabled: false,
-        scanned: false,
-        isLoading: !!artifactId,
-        isError: false,
-        ScanToolId: null,
-    })
+const SecurityTab = ({ ciPipelineId, artifactId, status, appIdFromParent, isJobCI }: SecurityTabType) => {
     const { appId } = useParams<{ appId: string }>()
     const { push } = useHistory()
-    async function callGetSecurityIssues() {
-        try {
-            const { result } = await getLastExecutionByArtifactId(appId ?? appIdFromParent, artifactId)
-            setSecurityData({
-                vulnerabilities: result.vulnerabilities,
-                lastExecution: result.lastExecution,
-                severityCount: result.severityCount,
-                scanEnabled: result.scanEnabled,
-                scanned: result.scanned,
-                isLoading: false,
-                isError: false,
-                ScanToolId: result.scanToolId,
-            })
-        } catch (error) {
-            // showError(error);
-            setSecurityData({
-                ...securityData,
-                isLoading: false,
-                isError: true,
-            })
-        }
-    }
+    const isSecurityScanV2Enabled = window._env_.ENABLE_RESOURCE_SCAN_V2 && isFELibAvailable()
 
-    function toggleCollapse() {
-        setIsCollapsed(!isCollapsed)
-    }
+    const [scanResultLoading, scanResultResponse, scanResultError] = useAsync(
```

Comment:
There seems to be code duplication, can we think of something like a hook or any other clean solution?

Reply:
both have different api for v1 one uses app id env id and is min api, other is appid and artifact id, so response type is also different